src/NHibernate/AdoNet/GenericBatchingBatcher.cs

```diff
@@ -183,7 +183,7 @@ public void Append(DbParameterCollection parameters)
 					_commandType = _batcher.CurrentCommand.CommandType;
 				}
 				
-				_sql.Add(_batcher.CurrentCommandSql.Copy());
+				_sql.Add(_batcher.GetSQL(_batcher.CurrentCommandSql.Copy()));
```

Comment:
Why `SqlString.Copy` call is required?

Reply:
In order to avoid modifying its parameters, which are exposed by `SqlString.GetParameters` method. The `Parameter` class is not immutable as `ParameterPosition` and `BackTrack` properties can be modified.

Reply:
Can't we call it once when final `SqlString` is created instead (`_sql.ToSqlString().Copy()`)?

Reply:
> Can't we call it once when final SqlString is created instead

As other batchers do not copy the `SqlString` prior calling the interceptor method, I've removed the `Copy` call to be consistent with them.